### PR TITLE
zephyr: update print messages for 64-bit atomics

### PR DESCRIPTION
**Pull request by cfriedt**
Atomics will use `long` rather than `int` so that they are 64-bit on 64-bit builds and 32-bit on 32-bit builds.

Required by zephyrproject-rtos/zephyr#39531
Originally at zephyrproject-rtos/sof#14

cc @lgirdwood @lyakh @kv2019i

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by plbossart** [comment]
> Can one of the admins verify this patch?
> 
> > reply **test this please** to run this test once

test this please

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

File: src/schedule/ll_schedule.c
Comment (lyakh) [inline src/schedule/ll_schedule.c:336]: strictly speaking, this file isn't used with Zephyr so these changes aren't fixing Zephyr builds directly, but it's true, that we need consistency here. OTOH using `%ld` with 32-bit **should** actually generate warnings like
```
/tmp/p.c: In function ‘main’:
/tmp/p.c:8:19: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 2 has type ‘int32_t’ {aka ‘int’} [-Wformat=]
    8 |  return printf("%ld\n", x);
      |                 ~~^     ~
      |                   |     |
      |                   |     int32_t {aka int}
      |                   long int
      |                 %d
```
in XTOS builds but since we don't use `__attribute__((__format__(printf,... ,...)))` with XTOS builds (so far) we shouldn't be getting these. So, so far this should be safe...

**Review by lyakh — COMMENTED** [review]
shall we wait until the Zephyr PR is merged before merging this?

**Comment by cfriedt** [comment]
> shall we wait until the Zephyr PR is merged before merging this?

No. The Zephyr fork is a downstream of this project, so the change needs to be merged here first, and then the Zephyr fork of sof needs to be updated, and then the west.yml ref needs to be updated in zephyrproject-rtos/zephyr#39531.

I realize it's a bit confusing, but the idea is to preserve the upstream / downstream relationship and have coherent git history.

**Review by dbaluta — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
Rerun CI, looks like it got stuck.

**Comment by cfriedt** [comment]
Just to reiterate, the zephyr build is expected to fail here.

**Comment by lgirdwood** [comment]
@kkarask @wszypelt looks like CI got stuck. Can you restart. Thanks !

**Comment by wszypelt** [comment]
@lgirdwood Done :-)

**Comment by marc-hb** [comment]
Follow-up in #5004 `zephyr/atomic.h: add a no-op cast: (long)atomic_read()`, please review.

**Pull request merged**
